### PR TITLE
update release CI and the self-managed-k8s setup to use latest bcc release

### PR DESCRIPTION
**Pull request by Ankurk99**
- update self-managed-k8s-selinux/setup.sh and contribution/self-managed-k8s/setup.sh to use latest bcc release
- update release workflow to use latest bcc release

### CODE REVIEW

**Review by nyrahul — APPROVED** [review]

**Comment by achrefbensaad** [comment]
As per this issue in the bcc repo [issue](https://github.com/iovisor/bcc/issues/3881), bcc will no longer support LLVM 7.
So the pipelines will break again once a new release of bcc is available.
Release v0.24.0 is the last release that will work for us.
Please see recommendations bellow.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ jobs:
           sudo apt-get update
           sudo apt-get -y install build-essential cmake bison flex git python3 python3-pip clang-9 libllvm9 llvm-9-dev libclang-9-dev zlib1g-dev libelf-dev libedit-dev libfl-dev
           pushd /tmp
-          git clone https://github.com/iovisor/bcc.git
```
Comment (achrefbensaad) [inline .github/workflows/release.yml:25]: `git -C /tmp/build/ clone https://github.com/iovisor/bcc.git --branch v0.24.0`

File: contribution/self-managed-k8s-selinux/setup.sh [hunk 2]
```diff
@@ -7,7 +7,8 @@ mkdir -p /tmp/build; cd /tmp/build
 
 # download bcc
 sudo dnf -y install git
-git -C /tmp/build/ clone https://github.com/iovisor/bcc.git
```
Comment (achrefbensaad) [inline contribution/self-managed-k8s-selinux/setup.sh:10]: `git -C /tmp/build/ clone https://github.com/iovisor/bcc.git --branch v0.24.0`

File: contribution/self-managed-k8s/setup.sh [hunk 3]
```diff
@@ -16,7 +16,8 @@ sudo apt-get update
 sudo rm -rf /tmp/build; mkdir -p /tmp/build; cd /tmp/build
 
 # download bcc
-git -C /tmp/build/ clone https://github.com/iovisor/bcc.git
```
Comment (achrefbensaad) [inline contribution/self-managed-k8s/setup.sh:19]: `git -C /tmp/build/ clone https://github.com/iovisor/bcc.git --branch v0.24.0`

**Review by achrefbensaad — COMMENTED** [review]
@Ankurk99 @nyrahul , Please give this a look

**Comment by Ankurk99** [comment]
> As per this issue in the bcc repo [issue](https://github.com/iovisor/bcc/issues/3881), bcc will no longer support LLVM 7. So the pipelines will break again once a new release of bcc is available. Release v0.24.0 is the last release that will work for us. Please see recommendations bellow.

I think instead of using a fixed version of bcc, we can use llvm 11.

**Comment by achrefbensaad** [comment]
> I think instead of using a fixed version of bcc, we can use llvm 11.

This will require a version bump for all related scripts (Vagrant, shell and github actions) to use ubuntu 21.04 as the bcc documentation introduces llvm11 starting from 21.04.

**Comment by Ankurk99** [comment]
@achrefbensaad Thanks for the analysis. @nyrahul What do you think about it?

**Comment by daemon1024** [comment]
I think we should restrict to v0.24.0 while we work on #38

**Comment by nyrahul** [comment]
Yeah we have to prioritize #38 anyways since BPF-LSM #484 depends on it to.

The point is, once we handle #38 .. we will remove dependency on BCC altogether.
Hence I am ok if we simply support BCC v0.24.0 at this point since we will lose BCC in the near future anyways.

**Comment by Ankurk99** [comment]
@achrefbensaad Would you be interested in fixing this issue?

**Comment by achrefbensaad** [comment]
@Ankurk99, Sure. Ill prepare a pull request for it.

**Comment by achrefbensaad** [comment]
this is the rollback [PR](https://github.com/kubearmor/KubeArmor/pull/637)

**Comment by viniciusd** [comment]
> > I think instead of using a fixed version of bcc, we can use llvm 11.
> 
> This will require a version bump for all related scripts (Vagrant, shell and github actions) to use ubuntu 21.04 as the bcc documentation introduces llvm11 starting from 21.04.

I don't think that's the case, their install docs are just outdated. llvm11 is available for previous Ubuntu versions as well

**Comment by Ankurk99** [comment]
@viniciusd Thanks for the feedback. I think we will stick to v0.24.0 'cause we have already planned to move to libbpf.

**Pull request merged**